src/Microsoft.Management.Deployment/PackageManager.cpp

```diff
@@ -317,34 +317,25 @@ namespace winrt::Microsoft::Management::Deployment::implementation
         return context;
     }
 
-    std::wstring GetQueueIdForInstall(winrt::Microsoft::Management::Deployment::CatalogPackage package, winrt::Microsoft::Management::Deployment::PackageCatalogInfo catalogInfo)
-    {
-        // Package Id + Catalog Id forms a unique identifier of a package for the purposes of the queue.
-        // Packages with the same id but from different catalogs can be installed at the same time.
-        std::wstring queueId;
-        queueId += package.Id();
-        queueId += std::wstring(L".");
-        queueId += catalogInfo.Id();
-        return std::move(queueId);
-    }
-
-    std::shared_ptr<Execution::OrchestratorQueueItem> GetQueueItemForPackage(winrt::Microsoft::Management::Deployment::CatalogPackage package, winrt::Microsoft::Management::Deployment::PackageCatalogInfo catalogInfo)
+    std::shared_ptr<Execution::OrchestratorQueueItem> GetExistingQueueItemForPackage(winrt::Microsoft::Management::Deployment::CatalogPackage package, winrt::Microsoft::Management::Deployment::PackageCatalogInfo catalogInfo)
     {
         std::shared_ptr<Execution::OrchestratorQueueItem> queueItem = nullptr;
         if (catalogInfo)
         {
             // If the caller has passed in the catalog they expect the package to have come from, then only look for an install from that catalog.
             // Fail if they've used a catalog that doesn't have an Id. This can currently happen for Info objects that come from PackageCatalogReference objects for REST catalogs.
-            THROW_HR_IF(APPINSTALLER_CLI_ERROR_INVALID_CL_ARGUMENTS, catalogInfo.Id().size() == 0);
-            queueItem = Execution::ContextOrchestrator::Instance().GetQueueItem(GetQueueIdForInstall(package, catalogInfo));
+            THROW_HR_IF(APPINSTALLER_CLI_ERROR_INVALID_CL_ARGUMENTS, catalogInfo.Id().empty());
+            queueItem = Execution::ContextOrchestrator::Instance().GetQueueItem(Execution::OrchestratorQueueItemFactory::CreateItemForInstall(package.Id().c_str(), catalogInfo.Id().c_str(), nullptr)->GetId().get());

```

Comment:
```suggestion
            queueItem = Execution::ContextOrchestrator::Instance().GetQueueItem(Execution::OrchestratorQueueItemFactory::CreateItemForInstall(std::wstring{ package.Id() }, std::wstring{ catalogInfo.Id() }, nullptr)->GetId().get());

```

And repeat below; this uses `hstring::operator std::wstring_view` to make the construction of the string not require a `wstrlen`.